tools/benchmark/src/MemoryTestRunner.ts

```diff
@@ -240,9 +267,11 @@ export interface MemoryTestArguments extends MochaExclusiveOptions, HookArgument
                     heapUsedArray.push(benchmarkStats.samples.after.memoryUsage[i].heapUsed
                                         - benchmarkStats.samples.before.memoryUsage[i].heapUsed);
                 }
-                heapUsedStats = getArrayStatistics(heapUsedArray);
+                heapUsedStats = getArrayStatistics(heapUsedArray, options.samplePercentageToUse);
             } while (benchmarkStats.runs < options.minSampleCount
                 || heapUsedStats.rme > options.maxRelativeMarginOfError);
+
+            benchmarkStats.stats = heapUsedStats;
```

Comment:
Should there be some sort of check to see that heapUsedStats is defined?

Reply:
The do-while loop above this is guaranteed to run and assign it before it exits, or if something fails in the loop then this line will be skipped.

Reply:
I was thinking a fail fast would be important here in case someone changes this in the future and somehow benchmarkStats is marked undefined. Feel free to resolve this conversation.